pkgs/applications/editors/kakoune/revert-uncaught-exceptions.patch

```diff
@@ -0,0 +1,73 @@
+From e2e3aee27980e10a419b37dd98263728954273c2 Mon Sep 17 00:00:00 2001
+From: Jason Felice <jason.m.felice@gmail.com>
+Date: Wed, 3 Jul 2019 12:31:58 -0400
+Subject: [PATCH] Revert "Fix BufferedWriter triggering std::terminate on
+ exception when writing"
+
+This reverts commit 262ef9b4e3d8bcd9252603778a12a3900024f23c.
+
+This requires the Mac OS C++ ABI to be >= 10.12, which is not currently
+the case on the Nix build system.
```

Comment:
Is upstream aware of this? If they don’t want to support older ABIs that’s fine, but seems a shame for me considering how small the change is.

Reply:
I mentioned it on IRC, but I'm not sure if @mawww was around.  The reverted commit actually reinstates a regression - a possible SEGV when saving some files in some cases.

Here's the part I don't quite understand, and I sent @matthewbauer an e-mail.  I think Nix [*wants* to be on 10.12](https://matthewbauer.us/blog/darwin-stdenv-update.html), and this was an oversight, something that is in progress, or something to be updated as demand requires.  If so, fixing *that* would solve this problem and not introduce the regression.

Kakoune has adopted c++17 as required for the C++ compiler, so I guessed that supporting < 10.12 (released 2016) wouldn't be interesting, especially since re-solving the regression looked complicated.

Here's my [build log](https://gist.github.com/eraserhd/9c630ef2b6844790c178d5f80a551339) for reference.